src/vsphere_cpi/lib/cloud/vsphere/cloud.rb

```diff
@@ -671,7 +671,21 @@ def generate_network_env(devices, networks, dvs_index)
         if device.kind_of?(Vim::Vm::Device::VirtualEthernetCard)
           backing = device.backing
           if backing.kind_of?(Vim::Vm::Device::VirtualEthernetCard::DistributedVirtualPortBackingInfo)
-            v_network_name = dvs_index[backing.port.portgroup_key]
+            network = @datacenter.mob.network.detect do |n|
```

Comment:
Can we split this into a series of `#select` and `#detect`s to make it easier to read?

Reply:
This can be split.